### PR TITLE
Support new <Banner /> component

### PR DESCRIPTION
**Pull request by kelsonic**
# Description

**Issues:**

https://github.com/department-of-veterans-affairs/va.gov-team/issues/29069
https://github.com/department-of-veterans-affairs/va.gov-team/issues/29209
https://github.com/department-of-veterans-affairs/va.gov-team/issues/28933
https://github.com/department-of-veterans-affairs/va.gov-team/issues/29071
https://github.com/department-of-veterans-affairs/va.gov-team/issues/28295
https://github.com/department-of-veterans-affairs/va.gov-team/issues/28296

**Related PRs:**

1. https://github.com/department-of-veterans-affairs/content-build/pull/485
1. https://github.com/department-of-veterans-affairs/vets-website/pull/18426
1. https://github.com/department-of-veterans-affairs/component-library/pull/147

This PR is part of a series of PRs to support the modified `EmergencyBanner` component and its switch to the new `Banner` component.

## Screenshots

image

image

image

## Acceptance Criteria

1. Renames `EmergencyBanner` to `Banner`.
1. Adds analytics to banner dismiss functionality.
1. Allows multiple banners to be dismissed.

### CODE REVIEW

File: src/components/Banner/Banner.jsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
+// Node modules.
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+// Relative imports.
+import AlertBox from '../AlertBox/AlertBox';
+
+const DISMISSED_BANNERS_KEY = 'DISMISSED_BANNERS';
+
+export class Banner extends Component {
```
Comment (cvalarida) [inline src/components/Banner/Banner.jsx:10]: Can you add some documentation to this component about how it manages state in `storage`? Specifically around the ability to dismiss multiple alerts. This change is what took me the longest to figure out.

**Review by cvalarida — COMMENTED** [review]
This look good. I'd just like to see documentation on the state management since that happens outside the scope of the component itself (_a la_ `localStorage`).

File: src/components/Banner/Banner.jsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  static propTypes = {
+    // A raw HTML string.
+    content: PropTypes.string.isRequired,
+    // Usually this is just window.localStorage or window.sessionStorage.
```
[the remark this answers is not in the document]
Comment (kelsonic) [reply src/components/Banner/Banner.jsx:14]: Definitely, happy to do that 🙂 Is the problem that storybook doesn't know how to read `// comment` comments? Can I just change the prop type comments to use this format: `/* comment */`?

File: src/components/Banner/Banner.jsx (hunk 1 above)
Comment (kelsonic) [reply src/components/Banner/Banner.jsx:10]: Sure! Where would be the best place for me to add the documentation, perhaps in the `Banner.mdx`? 🙂

File: src/components/Banner/Banner.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kelsonic) [reply src/components/Banner/Banner.jsx]: Updated, lmk if that's not what you meant! 🙂

File: src/components/Banner/Banner.jsx (hunk 1 above)
Comment (cvalarida) [reply src/components/Banner/Banner.jsx:10]: In a JSDoc right here. Storybook will pick it up. :grin:

:thinking: That `.mdx` file is I think no longer used. It used to be in the old documentation site, but now that Storybook is a thing, I don't think it's used.

File: src/components/Banner/Banner.jsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+export class Banner extends Component {
+  static propTypes = {
+    /**
+     * A raw HTML string.
+     * */
+    content: PropTypes.string.isRequired,
+    /**
+     * Usually this is just window.localStorage or window.sessionStorage.
+     * */
```
Comment (cvalarida) [inline src/components/Banner/Banner.jsx:18]: :sparkles: Awarded: 100 bonus points :sparkles:

File: src/components/Banner/Banner.jsx (hunk 1 above)
Comment (kelsonic) [reply src/components/Banner/Banner.jsx:10]: I updated the comment for `storage`, let me know if that's sufficient or if you were thinking of something else! 🙂 https://github.com/department-of-veterans-affairs/component-library/pull/147/commits/a000fdb9f980db38d8a30c5bd130c37fb70fc2ea

Comment (kelsonic) [reply src/components/Banner/Banner.jsx:10]: image

It looks like all the components on `master` still have the `.mdx` files btw, lmk if you want me to delete them and I'm happy to include that in this PR 👍

**Comment by kelsonic** [comment]
Hey @cvalarida would you be able to help me coordinate the deployment of this PR and the component library? I need to make sure [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/18426) is merged in at the same time and is using the latest component-library version that includes this PR's changes 🙂

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@department-of-veterans-affairs/component-library",
-  "version": "2.5.8",
+  "version": "2.5.9",
```
Comment (cvalarida) [inline package.json:3]: Because it introduces breaking changes (renaming the component), this is actually a major version bump.

Comment (kelsonic) [reply package.json]: Updated, thanks!

File: src/components/Banner/Banner.jsx (hunk 1 above)
Comment (cvalarida) [reply src/components/Banner/Banner.jsx:10]: Nah, we'll catch that in another PR since it's not part of the changes here. Thanks, though!

**Review by cvalarida — COMMENTED** [review]
I think the only other thing I'd like to see before approving—and sorry for not catching this earlier—is test coverage for the new features.

**Comment by kelsonic** [comment]
@cvalarida I started adding unit tests but then I realized that our scripts don't seem to work:
```
    "test-debug": "karma start testing/karma.conf.js --browsers Chrome",
    "test": "karma start testing/karma.conf.js --single-run",
    "test-watch": "karma start testing/karma.conf.js --auto-watch"
```
so I ended up not including them as they don't seem to ever be run anyways, except locally + manually

**Comment by cvalarida** [comment]
We _do_ run the tests in CI. Example: https://github.com/department-of-veterans-affairs/component-library/pull/147/checks?check_run_id=3488796979

**Comment by kelsonic** [comment]
@cvalarida ah ha! I started digging into it and it looks like it was just expecting my Banner.jsx file to end with .js instead, I'll add back the unit tests! 🙂

**Comment by kelsonic** [comment]
Unit tests added ✅

**Review by cvalarida — APPROVED** [review]
Looks great! :shipit:

**Comment by cvalarida** [comment]
Looks like it's still having some testing trouble, but once that gets fixed up and all the tests pass, it should be golden.

**Comment by kelsonic** [comment]
Thanks for the approval! 😃 Could you help me coordinate when I merge this in that it goes out at the same time as https://github.com/department-of-veterans-affairs/vets-website/pull/18426? I'm not familiar with how the component-library gets deployed

**Comment by cvalarida** [comment]
Looks like that's what the test has in it.

https://github.com/department-of-veterans-affairs/component-library/pull/147/files#diff-65c1f56871f5940752ded7ffa2f5b9958c87c3e80176de555cb75f2d14826cfdR6

**Comment by kelsonic** [comment]
Oh wow 🤦 haha thank you!

**Pull request merged**